mypy/test/data.py

```diff
@@ -490,7 +506,7 @@ def runtest(self) -> None:
         if self.skip:
             pytest.skip()
         update_data = self.config.getoption('--update-data', False)
-        self.parent.obj(update_data=update_data).run_case(self.obj)
+        self.parent.parent.obj(update_data=update_data).run_case(self.obj)
```

Comment:
This `.parent.parent.obj` gets kind of hard to trace through as a reader -- in fact, probably I should have made it more explicit the first time instead of saying `.parent.obj`.

Something like
```
data_file_collector = self.parent
class_collector = data_file_collector.parent
cls = class_collector.obj
cls(update_data=update_data).run_case(self.obj)
```
would be much clearer. Maybe a couple of those steps could be consolidated to make just one or two intermediate names rather than three, but I think zero intermediates is definitely too opaque.